java-frontend/src/main/java/org/sonar/java/se/checks/DivisionByZeroCheck.java

```diff
@@ -46,48 +47,43 @@
 @Rule(key = "S3518")
 public class DivisionByZeroCheck extends SECheck {
 
-  private enum DivByZeroStatus implements ObjectConstraint.Status {
-    ZERO {
-      @Override
-      public String valueAsString() {
+  private enum ZeroConstraint implements Constraint {
+    ZERO,
+    NON_ZERO;
+
+    @Override
+    public String valueAsString() {
+      if (this == ZeroConstraint.ZERO) {
         return "zero";
       }
-    },
-    NON_ZERO {
-      @Override
-      public String valueAsString() {
-        return "non-zero";
-      }
-    },
-    UNDETERMINED {
-      @Override
-      public String valueAsString() {
-        return "\"maybe-zero\"";
+      return "non-zero";
+    }
+
+    @Override
+    public boolean isValidWith(@Nullable Constraint constraint) {
+      return constraint == null || this == constraint;
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS1698)